llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -47,6 +47,11 @@ namespace llvm::AMDGPU {
 #include "AMDGPUGenSearchableTables.inc"
 } // namespace llvm::AMDGPU
 
+static cl::opt<bool> DisableDiffBasePtrMemClustering(
+    "amdgpu-disable-diff-baseptr-mem-clustering",
+    cl::desc("Disable clustering memory ops with different base pointers"),
+    cl::init(false), cl::Hidden);
```

Comment:
I would much prefer an "enable" flag. Otherwise you have to parse a double negative to understand what `-disable-thing=false` means.
```suggestion
static cl::opt<bool> EnableDiffBasePtrMemClustering(
    "amdgpu-disable-diff-baseptr-mem-clustering",
    cl::desc("Disable clustering memory ops with different base pointers"),
    cl::init(true), cl::Hidden);
```